plugins/modules/nomad_token.py

```diff
@@ -0,0 +1,285 @@
+    returned: always
+    type: dict
+    sample: {
+        "msg": "Acl token updated.",
+        "token": {
+            "accessor_id": "0d01c55f-8d63-f832-04ff-1866d4eb594e",
+            "create_index": 14,
+            "create_time": "2023-11-12T18:48:34.248857001Z",
+            "expiration_time": null,
+            "expiration_ttl": "",
+            "global": true,
+            "hash": "eSn8H8RVqh8As8WQNnC2vlBRqXy6DECogc5umzX0P30=",
+            "modify_index": 836,
+            "name": "devs",
+            "policies": [
+                "readonly"
+            ],
+            "roles": null,
+            "secret_id": "12e878ab-e1f6-e103-b4c4-3b5173bb4cea",
+            "type": "client"
+        }
+    }
+'''
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils.common.text.converters import to_native
+
+
+import_nomad = None
+
+
+try:
+    import nomad
+    import_nomad = True
+except ImportError:
+    import_nomad = False
+
+
+def get_token(name, nomad_client):
+
+    tokens = nomad_client.acl.get_tokens()
+
+    token = next((token for token in tokens
+                  if token.get('Name') == name), None)
+    return token
+
+
+def transform_response(nomad_response):
+
+    transformed_response = {
+        "accessor_id": nomad_response['AccessorID'],
+        "create_index": nomad_response['CreateIndex'],
+        "create_time": nomad_response['CreateTime'],
+        "expiration_ttl": nomad_response['ExpirationTTL'],
+        "expiration_time": nomad_response['ExpirationTime'],
+        "global": nomad_response['Global'],
+        "hash": nomad_response['Hash'],
+        "modify_index": nomad_response['ModifyIndex'],
+        "name": nomad_response['Name'],
+        "policies": nomad_response['Policies'],
+        "roles": nomad_response['Roles'],
+        "secret_id": nomad_response['SecretID'],
+        "type": nomad_response['Type']
+    }
+
+    return transformed_response
+
+
+def run():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            host=dict(required=True, type='str'),
+            port=dict(type='int', default=4646),
+            state=dict(required=True, choices=['present', 'absent']),
+            use_ssl=dict(type='bool', default=True),
+            timeout=dict(type='int', default=5),
+            validate_certs=dict(type='bool', default=True),
+            client_cert=dict(type='path'),
+            client_key=dict(type='path'),
+            namespace=dict(type='str'),
+            token=dict(type='str', no_log=True),
+            name=dict(type='str'),
+            token_type=dict(choices=['client', 'management', 'bootstrap'], default='client'),
+            policies=dict(type='list', elements='str', default=[]),
+            global_replicated=dict(type='bool', default=False),
+        ),
+        supports_check_mode=True,
+        required_one_of=[
+            ['name', 'token_type']
+        ]
```

Comment:
I think `name` is required if `token_type` is not `bootstrap`, right? If that's correct, you should add:
```suggestion
        ],
        required_if=[
            ('token_type', 'client', ('name', )),
            ('token_type', 'management', ('name', )),
        ],
```